integration_test/third_party_apps_data/applications/apache/metadata.yaml

```diff
@@ -1,4 +1,4 @@
-public_url: "https://cloud.google.com/stackdriver/docs/solutions/agents/ops-agent/third-party/apache"
+public_url: https://cloud.google.com/stackdriver/docs/solutions/agents/ops-agent/third-party/apache
 short_name: apache
```

Comment:
There are quite some changes in the per integration metadata.yaml files. Some are reordering. May i ask what's the reason to reorder them?🤔

Reply:
This was just the result of running the generate script on the existing yaml files -- the marshaller doesn't preserve ordering by default so the result is that things got out of order when reading and rewriting the file.

I would be fine with reverting all of these yaml changes in this PR since it is indeed quite a lot of changes. The caveat is that the next time someone runs the generate script, it'll change the ordering anyway.

Reply:
The thing with the current code is that it changes too many files unnecessarily. It will need reviewer carefully skim over one by one to see that nothing gets wrong.

This shall preserve the ordering: https://stackoverflow.com/questions/33639269/preserving-order-of-yaml-maps-using-go

Reply:
I think I tried `MapSlice` and ran into some walls with being able to continue using structured types with it. I also tried going deeper using v3 of the package which lets you access the underlying token stream but it still formats things slightly differently (arrays are indented in v3 but not in v2 and there's not enough control over it).

So for now I think a better solution is to just pull out all of these yaml changes to reduce the diff size.

Reply:
SGTM!
Didn't know in golang it needs this much work and still can't get it to work.. in python, there is oyaml that will just do the trick.

Another way i can think of is we can make generate.go only print out expected_metrics in the terminal, and letting the code submitter manually modify the metadata.yaml with the provided info showed in the terminal

Reply:
True, that would make the code much simpler too... I'll give that a try, good suggestion!